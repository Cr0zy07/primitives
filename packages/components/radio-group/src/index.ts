export { OkuRadioGroup } from './radio-group'
export { OkuRadioGroupItem } from './radio-group-item'
export { OkuRadioGroupIndicator } from './radio-group-indicator'

export type {
  RadioGroupProps,
  RadioGroupEmits,
  RadioGroupElement,
  RadioGroupNativeElement,
} from './props'

export type {
  RadioGroupItemProps,
  RadioGroupItemEmits,
  RadioGroupItemElement,
  RadioGroupItemNativeElement,
} from './props'

export type {
  RadioGroupIndicatorProps,
  RadioGroupIndicatorElement,
  RadioGroupIndicatorNativeElement,
} from './props'
