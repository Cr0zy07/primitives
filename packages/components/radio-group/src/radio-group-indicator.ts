import { defineComponent, h, mergeProps, reactive, toRefs } from 'vue'
import { reactiveOmit, useForwardRef } from '@oku-ui/use-composable'
import { OkuRadioIndicator } from './radio-indicator'
import { RADIO_GROUP_INDICATOR_NAME, radioGroupIndicatorProps, scopeRadioGroupProps, useRadioScope } from './props'
import type { RadioGroupIndicatorNativeElement } from './props'

const radioGroupIndicator = defineComponent({
  name: RADIO_GROUP_INDICATOR_NAME,
  components: {
    OkuRadioIndicator,
  },
  inheritAttrs: false,
  props: {
    ...radioGroupIndicatorProps.props,
    ...scopeRadioGroupProps,
  },
  emits: radioGroupIndicatorProps.emits,
  setup(props, { attrs }) {
    const { scopeOkuRadioGroup, ...indicatorProps } = toRefs(props)

    const _reactive = reactive(indicatorProps)
    const otherProps = reactiveOmit(_reactive, (key, _value) => key === undefined)

    const forwardedRef = useForwardRef()

    const radioScope = useRadioScope(scopeOkuRadioGroup.value)

    return () => h(OkuRadioIndicator, {
      ...radioScope,
      ...mergeProps(attrs, otherProps),
      ref: forwardedRef,
    })
  },
})

// TODO: https://github.com/vuejs/core/pull/7444 after delete
export const OkuRadioGroupIndicator = radioGroupIndicator as typeof radioGroupIndicator & (new () => { $props: RadioGroupIndicatorNativeElement })
