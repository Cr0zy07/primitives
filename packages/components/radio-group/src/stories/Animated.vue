<script setup lang="ts">
import { OkuRadioGroup, OkuRadioGroupIndicator, OkuRadioGroupItem } from '@oku-ui/radio-group'
import Label from './Label.vue'
</script>

<template>
  <Label>
    Favourite pet
    <OkuRadioGroup class="radio-group-root" default-value="1">
      <Label>
        <OkuRadioGroupItem class="radio-group-item" value="1">
          <OkuRadioGroupIndicator class="radio-group-animated-indicator radio-group-indicator" />
        </OkuRadioGroupItem>
        Cat
      </Label>{{ ' ' }}
      <Label>
        <OkuRadioGroupItem class="radio-group-item" value="2">
          <OkuRadioGroupIndicator class="radio-group-animated-indicator radio-group-indicator" />
        </OkuRadioGroupItem>
        Dog
      </Label>{{ ' ' }}
      <Label>
        <OkuRadioGroupItem class="radio-group-item" value="3">
          <OkuRadioGroupIndicator class="radio-group-animated-indicator radio-group-indicator" />
        </OkuRadioGroupItem>
        Rabbit
      </Label>
    </OkuRadioGroup>
  </Label>
</template>
