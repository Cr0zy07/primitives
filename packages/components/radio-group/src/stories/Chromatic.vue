<script setup lang="ts">
import { ref, watchEffect } from 'vue'
import { OkuRadioGroup, OkuRadioGroupIndicator, OkuRadioGroupItem } from '@oku-ui/radio-group'
import { OkuDirectionProvider } from '@oku-ui/direction'

const manualFocusRef = ref<HTMLDivElement | null>(null)

watchEffect(() => {
  manualFocusRef.value?.focus()
})
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Unset</h2>
  <OkuRadioGroup class="radio-group">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h2>Set</h2>
  <OkuRadioGroup class="radio-group" default-value="3">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h1>Controlled</h1>
  <h2>Unset</h2>
  <OkuRadioGroup class="radio-group" value="">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h2>Set</h2>
  <OkuRadioGroup class="radio-group" value="3">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h1>Disabled item</h1>
  <OkuRadioGroup class="radio-group">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2" disabled>
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h1>Disabled root</h1>
  <OkuRadioGroup class="radio-group" disabled>
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <!-- Not possible to set `disabled` back to `false` since it's set on the root (this item
    should still be disabled). -->
    <OkuRadioGroupItem class="radio-group-item" value="2" :disabled="false">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h1>All items disabled</h1>
  <OkuRadioGroup class="radio-group">
    <OkuRadioGroupItem class="radio-group-item" value="1" disabled>
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2" disabled>
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3" disabled>
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h1>Manual focus into group</h1>
  <OkuRadioGroup class="radio-group">
    <OkuRadioGroupItem ref="manualFocusRef" class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h1>Force mounted indicator</h1>
  <OkuRadioGroup class="radio-group">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" force-mount />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" force-mount />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" force-mount />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h1>Direction</h1>
  <h2>Prop</h2>
  <OkuRadioGroup class="radio-group" default-value="1" dir="rtl">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h2>Inherited</h2>
  <OkuDirectionProvider dir="rtl">
    <OkuRadioGroup class="radio-group" default-value="1">
      <OkuRadioGroupItem class="radio-group-item" value="1">
        <OkuRadioGroupIndicator class="radio-group-indicator" />
      </OkuRadioGroupItem>
      <OkuRadioGroupItem class="radio-group-item" value="2">
        <OkuRadioGroupIndicator class="radio-group-indicator" />
      </OkuRadioGroupItem>
      <OkuRadioGroupItem class="radio-group-item" value="3">
        <OkuRadioGroupIndicator class="radio-group-indicator" />
      </OkuRadioGroupItem>
    </OkuRadioGroup>
  </OkuDirectionProvider>

  <h1>State attributes</h1>
  <h2>Default</h2>
  <OkuRadioGroup class="radio-group-attr-styles" default-value="3">
    <OkuRadioGroupItem class="radio-group-attr-styles" value="1">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="2">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="3">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h2>Disabled item</h2>
  <OkuRadioGroup class="radio-group-attr-styles" default-value="3">
    <OkuRadioGroupItem class="radio-group-attr-styles" value="1">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="2" disabled>
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="3">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <OkuRadioGroup class="radio-group-attr-styles" default-value="2">
    <OkuRadioGroupItem class="radio-group-attr-styles" value="1">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="2" disabled>
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="3">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h2>Disabled root</h2>
  <OkuRadioGroup class="radio-group-attr-styles" default-value="3" disabled>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="1">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="2">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="3">
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>

  <h2>All items disabled</h2>
  <OkuRadioGroup class="radio-group-attr-styles" default-value="3">
    <OkuRadioGroupItem class="radio-group-attr-styles" value="1" disabled>
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="2" disabled>
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-attr-styles" value="3" disabled>
      <OkuRadioGroupIndicator class="radio-group-attr-styles" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>
</template>
