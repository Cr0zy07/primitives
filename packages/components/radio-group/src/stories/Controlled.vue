<script setup lang="ts">
import { ref } from 'vue'
import { OkuRadioGroup, OkuRadioGroupIndicator, OkuRadioGroupItem } from '@oku-ui/radio-group'

const value = ref('2')
</script>

<template>
  <OkuRadioGroup class="radio-group" :value="value" @value-change="value = $event">
    <OkuRadioGroupItem class="radio-group-item" value="1">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="2">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
    <OkuRadioGroupItem class="radio-group-item" value="3">
      <OkuRadioGroupIndicator class="radio-group-indicator" />
    </OkuRadioGroupItem>
  </OkuRadioGroup>
</template>
