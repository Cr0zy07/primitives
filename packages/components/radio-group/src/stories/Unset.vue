<script setup lang="ts">
import { OkuRadioGroup, OkuRadioGroupIndicator, OkuRadioGroupItem } from '@oku-ui/radio-group'
import Label from './Label.vue'
</script>

<template>
  <Label>
    Favourite pet
    <OkuRadioGroup class="radio-group-root">
      <Label>
        <OkuRadioGroupItem class="radio-group-item" value="1">
          <OkuRadioGroupIndicator class="radio-group-indicator" />
        </OkuRadioGroupItem>
        Cat
      </Label>{{ ' ' }}
      <Label>
        <OkuRadioGroupItem class="radio-group-item" value="2" disabled>
          <OkuRadioGroupIndicator class="radio-group-indicator" />
        </OkuRadioGroupItem>
        Dog
      </Label>{{ ' ' }}
      <Label>
        <OkuRadioGroupItem class="radio-group-item" value="3">
          <OkuRadioGroupIndicator class="radio-group-indicator" />
        </OkuRadioGroupItem>
        Rabbit
      </Label>
    </OkuRadioGroup>
  </Label>
</template>
