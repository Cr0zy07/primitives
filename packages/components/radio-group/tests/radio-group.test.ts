import { defineComponent, ref, watchEffect } from 'vue'
import { enableAutoUnmount, mount, shallowMount } from '@vue/test-utils'
import type { DOMWrapper, VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuRadioGroup, OkuRadioGroupIndicator, OkuRadioGroupItem } from '../src'

import Styled from '../src/stories/Styled.vue'
import Controlled from '../src/stories/Controlled.vue'
import Unset from '../src/stories/Unset.vue'
import WithinForm from '../src/stories/WithinForm.vue'
import Animated from '../src/stories/Animated.vue'
import Chromatic from '../src/stories/Chromatic.vue'

enableAutoUnmount(afterEach)

const INDICATOR_TEST_ID_1 = 'radiogroup-indicator-1'
const INDICATOR_TEST_ID_2 = 'radiogroup-indicator-2'

const onValueChange = vi.fn()

const RadioGroupTest = defineComponent({
  components: {
    OkuRadioGroup,
    OkuRadioGroupItem,
    OkuRadioGroupIndicator,
  },
  props: {
    defaultValue: String,
    value: String,
    disabled: Boolean,
  },
  setup() {
    const containerRef = ref<HTMLDivElement | null>(null)
    watchEffect(() => {
      // We use the `hidden` attribute to hide the nested input from both sighted users and the
      // accessibility tree. This is perfectly valid so long as users don't override the display of
      // `hidden` in CSS. Unfortunately axe doesn't recognize this, so we get a violation because the
      // input doesn't have a label. This adds an additional `aria-hidden` attribute to the input to
      // get around that.
      // https://developer.paciellogroup.com/blog/2012/05/html5-accessibility-chops-hidden-and-aria-hidden/
      containerRef.value?.querySelector('input')?.setAttribute('aria-hidden', 'true')
    })

    return {
      containerRef,
      INDICATOR_TEST_ID_1,
      INDICATOR_TEST_ID_2,
      onValueChange,
    }
  },
  template: `
    <div ref="containerRef">
      <OkuRadioGroup v-bind="$attrs" @value-change="onValueChange">
        <OkuRadioGroupItem aria-label="basic radiogroup" value="1">
          <OkuRadioGroupIndicator :data-testid="INDICATOR_TEST_ID_1" />
        </OkuRadioGroupItem>
        <OkuRadioGroupItem aria-label="basic radiogroup" value="2">
          <OkuRadioGroupIndicator :data-testid="INDICATOR_TEST_ID_2" />
        </OkuRadioGroupItem>
      </OkuRadioGroup>
    </div>
  `,
})

globalThis.ResizeObserver = class ResizeObserver {
  cb: any
  constructor(cb: any) {
    this.cb = cb
  }

  observe() {
    this.cb([{ borderBoxSize: { inlineSize: 0, blockSize: 0 } }])
  }

  unobserve() { }
  disconnect() { }
}

describe('okuRadioGroup', () => {
  let wrapper: VueWrapper
  let radio_1: DOMWrapper<Element>
  let radio_2: DOMWrapper<Element>
  // let indicator_2: DOMWrapper<Element>

  beforeEach(() => {
    wrapper = mount(RadioGroupTest, {
      attachTo: document.body,
    })

    radio_1 = wrapper.find('[value="1"]')
    radio_2 = wrapper.find('[value="2"]')

    // indicator_2 = wrapper.find(`[data-testid="${INDICATOR_TEST_ID_2}"]`)
  })

  it('should render OkuRadioGroup correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()

    expect(shallowMount(OkuRadioGroup).html()).toMatchSnapshot()
  })

  it('should render OkuRadioGroupItem correctly', () => {
    const spy = vi.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuRadioGroupItem)

    expect(() => wrapper()).toThrowErrorMatchingSnapshot()

    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuRadioGroup)" not found.')
  })

  it('should render OkuRadioGroupIndicator correctly', () => {
    const spy = vi.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuRadioGroupIndicator)

    expect(() => wrapper()).toThrowErrorMatchingSnapshot()

    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuRadio)" not found.')
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  describe('when clicking the radio 2', () => {
    beforeEach(async () => {
      radio_2.trigger('click')
      // indicator_2 = wrapper.find(`[data-testid="${INDICATOR_TEST_ID_2}"]`)
    })

    it('should render a visible indicator', () => {
      expect(wrapper.find(`[data-testid="${INDICATOR_TEST_ID_2}"]`).isVisible()).toBe(true)
    })

    describe('and clicking the radio 1', () => {
      beforeEach(async () => {
        radio_1.trigger('click')
      })

      it('should remove the indicator', () => {
        expect(wrapper.find(`[data-testid="${INDICATOR_TEST_ID_2}"]`).exists()).toBe(false)
      })
    })
  })

  describe('given a disabled Radio', () => {
    let wrapper: VueWrapper

    beforeEach(() => {
      wrapper = mount(RadioGroupTest, {
        props: {
          disabled: false,
        },
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })
  })

  describe('given an uncontrolled `checked` RadioGroup', () => {
    let wrapper: VueWrapper
    let radio_1: DOMWrapper<Element>
    let indicator_2: DOMWrapper<Element>

    beforeEach(() => {
      wrapper = mount(RadioGroupTest, {
        props: {
          defaultValue: '1',
        },
        attachTo: document.body,
      })

      radio_1 = wrapper.find('[value="1"]')
      radio_2 = wrapper.find('[value="2"]')

      indicator_2 = wrapper.find(`[data-testid="${INDICATOR_TEST_ID_2}"]`)
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    describe('when clicking the radio', () => {
      beforeEach(async () => {
        radio_1.trigger('click')
      })

      it('should remove the indicator', () => {
        expect(indicator_2.exists()).toBe(false)
      })

      it('should call `onValueChange` event', () => {
        expect(onValueChange).toHaveBeenCalled()
      })
    })

    describe('given a controlled `checked` RadioGroup', () => {
      let wrapper: VueWrapper
      let radio_1: DOMWrapper<Element>

      beforeEach(() => {
        wrapper = mount(RadioGroupTest, {
          props: {
            value: '1',
          },
          attachTo: document.body,
        })

        radio_1 = wrapper.find('[value="1"]')
      })

      describe('when clicking the radio 1', () => {
        beforeEach(() => {
          radio_1.trigger('click')
        })

        it('should call `onValueChange` event', () => {
          expect(onValueChange).toHaveBeenCalled()
        })
      })
    })
  })
})

describe('okuRadioGroup Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = shallowMount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('controlled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Controlled>>

    beforeEach(async () => {
      wrapper = shallowMount(Controlled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('unset', () => {
    let wrapper: VueWrapper<InstanceType<typeof Unset>>

    beforeEach(async () => {
      wrapper = shallowMount(Unset, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('withinForm', () => {
    let wrapper: VueWrapper<InstanceType<typeof WithinForm>>

    beforeEach(async () => {
      wrapper = shallowMount(WithinForm, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('animated', () => {
    let wrapper: VueWrapper<InstanceType<typeof Animated>>

    beforeEach(async () => {
      wrapper = shallowMount(Animated, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('chromatic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Chromatic>>

    beforeEach(async () => {
      wrapper = shallowMount(Chromatic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
