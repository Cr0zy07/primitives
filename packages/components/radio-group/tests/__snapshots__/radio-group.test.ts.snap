// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuRadioGroup > should render OkuRadioGroup correctly 1`] = `
"<div>
  <div role=\\"radiogroup\\" aria-required=\\"false\\" dir=\\"ltr\\" tabindex=\\"0\\" style=\\"outline: none;\\"><button type=\\"button\\" role=\\"radio\\" aria-checked=\\"false\\" data-state=\\"unchecked\\" aria-label=\\"basic radiogroup\\" tabindex=\\"-1\\" data-oku-collection-item=\\"\\" value=\\"1\\">
      <!---->
    </button>
    <!----><button type=\\"button\\" role=\\"radio\\" aria-checked=\\"false\\" data-state=\\"unchecked\\" aria-label=\\"basic radiogroup\\" tabindex=\\"-1\\" data-oku-collection-item=\\"\\" value=\\"2\\">
      <!---->
    </button>
    <!---->
  </div>
</div>"
`;

exports[`okuRadioGroup > should render OkuRadioGroup correctly 2`] = `"<oku-roving-focus-group-stub scopeokurovingfocusgroup=\\"[object Object]\\" dir=\\"ltr\\" loop=\\"true\\" aschild=\\"true\\"></oku-roving-focus-group-stub>"`;

exports[`okuRadioGroup > should render OkuRadioGroupIndicator correctly 1`] = `"\`OkuRadioIndicator\` must be used within \`OkuRadio\`"`;

exports[`okuRadioGroup > should render OkuRadioGroupItem correctly 1`] = `"\`OkuRadioGroupItem\` must be used within \`OkuRadioGroup\`"`;

exports[`okuRadioGroup Stories > animated > should render correctly 1`] = `"<label-stub></label-stub>"`;

exports[`okuRadioGroup Stories > chromatic > should render correctly 1`] = `
"<h1>Uncontrolled</h1>
<h2>Unset</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h2>Set</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" defaultvalue=\\"3\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h1>Controlled</h1>
<h2>Unset</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\" value=\\"\\"></oku-radio-group-stub>
<h2>Set</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\" value=\\"3\\"></oku-radio-group-stub>
<h1>Disabled item</h1>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h1>Disabled root</h1>
<oku-radio-group-stub required=\\"false\\" disabled=\\"true\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h1>All items disabled</h1>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h1>Manual focus into group</h1>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h1>Force mounted indicator</h1>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h1>Direction</h1>
<h2>Prop</h2>
<oku-radio-group-stub dir=\\"rtl\\" required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" defaultvalue=\\"1\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
<h2>Inherited</h2>
<direction-provider-stub dir=\\"rtl\\"></direction-provider-stub>
<h1>State attributes</h1>
<h2>Default</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" defaultvalue=\\"3\\" aschild=\\"false\\" class=\\"radio-group-attr-styles\\"></oku-radio-group-stub>
<h2>Disabled item</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" defaultvalue=\\"3\\" aschild=\\"false\\" class=\\"radio-group-attr-styles\\"></oku-radio-group-stub>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" defaultvalue=\\"2\\" aschild=\\"false\\" class=\\"radio-group-attr-styles\\"></oku-radio-group-stub>
<h2>Disabled root</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"true\\" loop=\\"true\\" defaultvalue=\\"3\\" aschild=\\"false\\" class=\\"radio-group-attr-styles\\"></oku-radio-group-stub>
<h2>All items disabled</h2>
<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" defaultvalue=\\"3\\" aschild=\\"false\\" class=\\"radio-group-attr-styles\\"></oku-radio-group-stub>"
`;

exports[`okuRadioGroup Stories > controlled > should render correctly 1`] = `"<oku-radio-group-stub required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\" value=\\"2\\"></oku-radio-group-stub>"`;

exports[`okuRadioGroup Stories > styled > should render correctly 1`] = `"<label-stub></label-stub>"`;

exports[`okuRadioGroup Stories > unset > should render correctly 1`] = `"<label-stub></label-stub>"`;

exports[`okuRadioGroup Stories > withinForm > should render correctly 1`] = `
"<form>
  <fieldset>
    <legend>optional value: </legend>
    <oku-radio-group-stub name=\\"optional\\" required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
  </fieldset><br><br>
  <fieldset>
    <legend>required value: </legend>
    <oku-radio-group-stub name=\\"required\\" required=\\"true\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
  </fieldset><br><br>
  <fieldset>
    <legend>stop propagation value: </legend>
    <oku-radio-group-stub name=\\"stopprop\\" required=\\"false\\" disabled=\\"false\\" loop=\\"true\\" aschild=\\"false\\" class=\\"radio-group\\"></oku-radio-group-stub>
  </fieldset><br><br><button>Submit</button>
</form>"
`;
